scalafmt-tests/src/test/scala/org/scalafmt/config/ConfigDialectOverrideTest.scala

```diff
@@ -0,0 +1,123 @@
+package org.scalafmt.config
+
+import munit.FunSuite
+
+class ConfigDialectOverrideTest extends FunSuite {
+
+  // toplevelSeparator is never actually used,
+  // but as the only non-boolean Dialect value it makes for a good test
+  test("dialect override - non boolean setting") {
+    ScalafmtConfig.fromHoconString(
+      """|
+         |runner.dialectOverride.toplevelSeparator = ">"
+         |runner.dialect = scala213
+         |""".stripMargin,
+    ).get
```

Comment:
I'd add an assert here. even though the setting isn't used, you want to test the overrides.

Reply:
The problem here is that the topLevelSeparator is deprecated and unimplemented (withToplevelSeparator(...) in the Dialect always returns `this`), so the idea here was to mostly check the Map method content not crashing in any way for the non boolean types (this is the only one for now). I can remove this test if it's too confusing/unnecessary

Reply:
understood, thank you, let's keep as-is.